Source/WebCore/Modules/indexeddb/IDBRequest.h

```diff
@@ -105,6 +105,7 @@ class IDBRequest : public EventTarget, public IDBActiveDOMObject, public ThreadS
     IndexedDB::IndexRecordType requestedIndexRecordType() const;
 
     ScriptExecutionContext* scriptExecutionContext() const final;
+    using IDBActiveDOMObject::protectedScriptExecutionContext;
```

Comment:
Why are all these `using` needed? Don't we inherit the public member function from the base class without an explicit `using`? Usually this is only needed if overloading causes it to be hidden, if we want to change visibility, or to resolve ambiguity if there is more than one inherited function with the same name.

Reply:
Unfortunately they are in order to resolve ambiguity because those classes inherit `protectedScriptExecutionContext()` from both `EventTarget` and `ContextDestructionObserver` (often via `ActiveDOMObject`). I resolve ambiguity by calling the `ContextDestructionObserver` / `ActiveDOMObject` once since the `EventTarget` one would end up calling a virtual function.